web/components/bet-button.tsx

```diff
@@ -64,11 +71,59 @@ export default function BetButton(props: {
         <SimpleBetPanel
           className={betPanelClassName}
           contract={contract}
-          selected="YES"
           onBetSuccess={() => setOpen(false)}
           hasShares={hasYesShares || hasNoShares}
         />
       </Modal>
     </>
   )
 }
+
+export function BinaryMobileBetting(props: { contract: BinaryContract }) {
+  const { contract } = props
+  const user = useUser()
+  if (user) {
+    return (
+      <>
+        <SignedInBinaryMobileBetting contract={contract} user={user} />
+      </>
+    )
+  } else {
+    return (
+      <>
+        <BetSignUpPrompt className="w-full" />
+      </>
+    )
+  }
+}
+
+export function SignedInBinaryMobileBetting(props: {
+  contract: BinaryContract
+  user: User
+}) {
+  const { contract, user } = props
+  const unfilledBets = useUnfilledBets(contract.id) ?? []
+
+  return (
+    <>
+      <Col className="w-full gap-2 px-1">
+        <Col>
+          <BuyPanel
+            hidden={false}
+            contract={contract as CPMMBinaryContract}
```

Comment:
You might want to type the contract field in props with `CPMMBinaryContract` so that no casting is required (and the props are accurate).